### ISSUE TITLE
Notebook testing

### ISSUE THREAD

**Issue by bryevdv** [bokeh/bokeh#9742]
We have been without proper CI testing for notebook and JupyterLab for too long. We need to make CI testing for these a top priority after 2.0.

**Comment by jbednar** [comment]
Agreed. The HoloViews team has had a long and painful history with notebook testing, much of which can be inferred from comments in https://github.com/holoviz/holoviews/pull/429 and https://github.com/holoviz/holoviews/issues/341. We started out with strict notebook testing, capturing the outputs from each cell, storing those as references, and then comparing to them, but we eventually gave up because the process of updating them was too cumbersome compared to the number of actual problems they found. Nowadays all we do is [nbsmoke](https://github.com/pyviz-dev/nbsmoke), which checks only that they run and do not contain certain types of lint. I.would hope the situation has improved in recent years and I'd love to get our projects back on output testing, because there have been too many cases where we had to detect visually that there were errors on the already-published websites. I imagine @jlstevens and @philippjfr have additional background they could share.
